csvcubed/csvcubed/readers/cubeconfig/v1/columnschema.py

```diff
@@ -78,20 +90,48 @@ def map_to_new_qb_dimension(
         )
         # The NewQbCodeList and Concepts are populated in the NewQbDimension.from_data() call
         # the _get_code_list method overrides the code_list if required.
-        new_dimension.code_list = self._get_code_list(new_dimension, csv_column_title)
+        new_dimension.code_list = self._get_code_list(
+            new_dimension, csv_column_title, cube_config_path=config_path
+        )
         return new_dimension
 
     def _get_code_list(
-        self, new_dimension: NewQbDimension, csv_column_title: str
+        self,
+        new_dimension: NewQbDimension,
+        csv_column_title: str,
+        cube_config_path: Optional[Path],
     ) -> Optional[QbCodeList]:
-
         if isinstance(self.code_list, str):
             if looks_like_uri(self.code_list):
                 return ExistingQbCodeList(self.code_list)
+            # The following elif is for cube config v1.1. This also requires the user to define the configuration in the build command, and therefore cube_config_path.
+            elif cube_config_path and code_list_config_json_exists(
+                Path(self.code_list), cube_config_path.parent
+            ):
+                code_list_config_path = (
+                    cube_config_path.parent / self.code_list
+                ).resolve()
+                _logger.info(
+                    f"Loading code list from local file path: {code_list_config_path}"
+                )
 
+                code_list_config, code_list_config_dict = CodeListConfig.from_json_file(
+                    code_list_config_path
+                )
+                schema = load_resource(code_list_config.schema)
+
+                code_list_schema_validation_errors = validate_dict_against_schema(
+                    value=code_list_config_dict, schema=schema
+                )
+                for error_msg in code_list_schema_validation_errors:
+                    _logger.warn(error_msg)
```

Comment:
I think we should try and pass this information back up the call stack to the `build` command if possible and let that write them out to the appropriate log.

Reply:
I might need to check this with you. I will give you a shout so we can have a co-screen.